components/list/list-item-drag-handle.js

```diff
@@ -212,52 +167,103 @@ class ListItemDragHandle extends LocalizeCoreElement(RtlMixin(LitElement)) {
 		}));
 	}
 
-	_dispatchActionDown() {
-		this._dispatchAction(dragActions.down);
+	async _doAction(action) {
+		this._dispatchAction(action);
+		const cell = findComposedAncestor(this, (parent) =>  parent.hasAttribute && parent.hasAttribute('draggable'));
+		if (cell) await cell.updateComplete;
+		await this.updateComplete;
+		this.focus();
+		this._movingElement = false;
+	}
+
+	_onDraggerButtonClick() {
+		this.activateKeyboardMode();
 	}
 
-	_dispatchActionUp() {
-		this._dispatchAction(dragActions.up);
+	_onDraggerButtonKeydown(e) {
+		if (e.keyCode !== keyCodes.ENTER && e.keyCode !== keyCodes.SPACE) return;
+		e.preventDefault();
+		this.activateKeyboardMode();
 	}
 
-	async _onActiveKeyboard(e) {
-		if (!this._keyboardActive) {
+	_onKeyboardButtonFocusIn() {
+		if (hasDisplayedKeyboardTooltip) return;
+		this._displayKeyboardTooltip = true;
+		hasDisplayedKeyboardTooltip = true;
+	}
+
+	_onKeyboardButtonFocusOut(e) {
+		this._displayKeyboardTooltip = false;
+		if (this._movingElement) {
+			this._movingElement = false;
+			e.stopPropagation();
+			e.preventDefault();
 			return;
 		}
+		this._keyboardActive = false;
+		this._dispatchAction(dragActions.save);
+		e.stopPropagation();
+	}
+
+	async _onMoveButtonAction(e) {
+
 		let action = null;
-		switch (e.keyCode) {
-			case keyCodes.UP:
+		switch (e.detail.action) {
+			case moveActions.up:
 				this._movingElement = true;
 				action = dragActions.up;
 				this.updateComplete.then(() => this.blur()); // tell screenreaders to refocus
 				break;
-			case keyCodes.DOWN:
+			case moveActions.down:
 				this._movingElement = true;
 				action = dragActions.down;
 				break;
-			case keyCodes.HOME:
+			case moveActions.right:
+				this._movingElement = true;
+				action = (this.dir === 'rtl' ? dragActions.unnest : dragActions.nest);
```

Comment:
Would it make sense for `button-move` to handle the RTL flip for consumers? If so, maybe we use directional agnostic terms like start/end or back/forward.

Reply:
I don't _think_ so... the `d2l-button-move`'s action detail reflects the direction the user selected exactly, and the consumer may or may not want to invert the operation. In the case of list-item, we do want it inverted. However if this were being used to move a floating container on the page, we'd want the direction it's moved to match the arrow keys being pressed, so we wouldn't want it inverted in that case.

Reply:
Right. I guess if we ended up with a lot more consumers it could end up being an option they pass in.